hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/cache/EntityMapWarmer.java

```diff
@@ -0,0 +1,258 @@
+    private static EntityMapWarmer instance;
+    // Note: these suppliers need to be stored here as they are (instead of storing what
+    // the supplier references) in order for this class to get the latest copies of the
+    // underlying data structures
+    private final Supplier<AccountStorageAdapter> accountsStorageAdapter;
+    private final Supplier<UniqueTokenMapAdapter> nftsAdapter;
+    private final Supplier<TokenRelStorageAdapter> tokenRelsAdapter;
+    private final ThreadPoolExecutor threadpool;
+
+    private EntityMapWarmer(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        this(
+                accountsStorageAdapter,
+                nftsAdapter,
+                tokenRelsAdapter,
+                new ThreadPoolExecutor(
+                        globalDynamicProperties.cacheCryptoTransferWarmThreads(),
+                        globalDynamicProperties.cacheCryptoTransferWarmThreads(),
+                        0L,
+                        TimeUnit.MILLISECONDS,
+                        new LinkedBlockingQueue<>()));
+    }
+
+    @VisibleForTesting
+    EntityMapWarmer(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            ThreadPoolExecutor threadpool) {
+        this.accountsStorageAdapter = accountsStorageAdapter;
+        this.nftsAdapter = nftsAdapter;
+        this.tokenRelsAdapter = tokenRelsAdapter;
+        this.threadpool = threadpool;
+    }
+
+    public static EntityMapWarmer getInstance(
+            Supplier<AccountStorageAdapter> accountsStorageAdapter,
+            Supplier<UniqueTokenMapAdapter> nftsAdapter,
+            Supplier<TokenRelStorageAdapter> tokenRelsAdapter,
+            GlobalDynamicProperties globalDynamicProperties) {
+        if (instance == null) {
+            instance =
+                    new EntityMapWarmer(accountsStorageAdapter, nftsAdapter, tokenRelsAdapter, globalDynamicProperties);
+        }
+        return instance;
+    }
+
+    /**
+     * Warms the cache for select entities found in the given {@code Round}. Any entities not
+     * found will, of course, not be warmed, but instead will essentially be ignored.
+     *
+     * @param round the round to warm the cache for
+     */
+    public void warmCache(Round round) {
+        removeAllTasks();
+        threadpool.execute(() -> doWarmups(round));
+    }
+
+    // "warmTokenObjs" as in "warm the token and its associated token relation objects"
+    // for both the sending and receiving accounts
+    private void warmTokenObjs(AccountID sender, AccountID receiver, long tokenNum, long senderSerialNum) {
+        if (accountsStorageAdapter.get().areOnDisk()
```

Comment:
Since these `are/Is*()` methods will always return the same answer, we could compute and store it once.

Ofc this should be inlined, so performance won't change, but should make semantics a bit clearer.